.cirrus.yml

```diff
@@ -0,0 +1,272 @@
+  start_xvfb_background_script:
+    - /etc/init.d/xvfb start
+  script:
+    - sleep 5
+    - source cirrus-env QA
+    - export packageVersion=$(npm pkg get version | tr -d \")
+    - npm run compile
+    - npm run prepare
+    - npm run test-cov
+    - .cirrus/cirrus_sonar_analysis_npm
+
+  always:
+    unit_tests_artifacts:
+      path: '**/alltests.xml'
+      format: junit
+
+mend_scan_task:
+  depends_on:
+    - test_linux
+  <<: *ONLY_MAIN_BRANCHES
+  eks_container:
+    <<: *CONTAINER_DEFINITION
+  <<: *NODE_MODULES_CACHE_TEMPLATE
+  env:
+    WS_APIKEY: VAULT[development/kv/data/mend data.apikey]
+    JDK_VERSION: "11"
+  whitesource_script:
+    - source cirrus-env QA # read build number
+    - export PROJECT_VERSION=$(npm pkg get version | tr -d \" | sed "s|-SNAPSHOT||g")
+    - source ws_scan.sh
+  allow_failures: "true"
+  always:
+    ws_artifacts:
+      path: "whitesource/**/*"
+
+qa_task:
+  #  name: Run ITs
+  depends_on:
+    - test_linux
+    - test_windows
+    - mend_scan
+  <<: *ONLY_IF_EXCEPT_NIGHTLY
+  eks_container:
+    <<: *CONTAINER_DEFINITION
+    cpu: 4
+    memory: 4G
+  <<: *NODE_MODULES_CACHE_TEMPLATE
+  matrix:
+    - env:
+        QA_CATEGORY: oldest_supported
+        VSCODE_VERSION: '1.73.0'
+    - env:
+        QA_CATEGORY: stable
+        VSCODE_VERSION: 'stable'
+    - env:
+        QA_CATEGORY: insiders
+        VSCODE_VERSION: 'insiders'
+  env:
+    DISPLAY: ":10"
+    RECORD_FILENAME: "${CIRRUS_WORKING_DIR}/record_vscode_${QA_CATEGORY}.mp4"
+  start_xvfb_background_script:
+    - /etc/init.d/xvfb start
+  prepare_tests_script:
+    - sleep 5 # ensure xvfb is up
+    - source cirrus-env QA
+    - export PROJECT_VERSION=$(npm pkg get version | tr -d \" | sed "s|-SNAPSHOT|+$BUILD_NUMBER|g")
+    - echo "Download staged vsix for ${PROJECT_VERSION}"
+    - mvn org.apache.maven.plugins:maven-dependency-plugin:3.1.1:copy -B --settings ~/.m2/settings.xml -Denable-repo=qa -Dartifact=org.sonarsource.sonarlint.vscode:sonarlint-vscode:$PROJECT_VERSION:vsix -DoutputDirectory=$CIRRUS_WORKING_DIR
+    - cd its
+    - npm ci
+  start_screen_recording_background_script:
+    - ffmpeg -f x11grab -video_size 1280x960 -i ${DISPLAY} -codec:v libx264 -r 12 ${RECORD_FILENAME}
+  test_script:
+    - source cirrus-env QA
+    - cd its
+    - npm test
+  stop_screen_recording_script:
+    - pkill -SIGINT -f ffmpeg
+    - while pgrep ffmpeg >/dev/null; do sleep 1; done
+    - /etc/init.d/xvfb stop
+  dump_code_logs_script:
+    - .cirrus/dump_code_logs
+  always:
+    its_tests_artifacts:
+      path: '**/*tests.xml'
+      format: junit
+    its_test_recording_artifacts:
+      path: ${RECORD_FILENAME}
+
+promote_task:
+  env:
+    ARTIFACTORY_ACCESS_TOKEN: VAULT[development/artifactory/token/${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}-promoter access_token]
+    PROMOTE_URL: VAULT[development/kv/data/promote data.url]
+    ARTIFACTS: "org.sonarsource.sonarlint.vscode:sonarlint-vscode:vsix"
+  depends_on:
+    - qa
```

Comment:
I think it would be more correct that this task depends on `qa`, `test_windows` and `test_linux`.